docs/troubleshooting.md

```diff
@@ -98,6 +98,142 @@ for p in $(oc get pods -l component=es -o jsonpath={.items[*].metadata.name}); d
   oc exec -c elasticsearch $ES_POD -- touch /opt/app-root/src/init_failures;  \
 done
 ```
+### Elasticsearch performance degrades around 00:00 UTC
+This situation is a result of Elasticsearch processing bulk index requests from all the nodes in the OKD cluster for
+logs for the new day.  Creation of a significant number of new indices simultaneously can consume significant resources when there are a large number of projects.  Periodically pre-creating the next day's indices can reduce the load on ElasticSearch when rolling over to the new day:
+```
+apiVersion: v1
+kind: Template
+metadata:
+  name: indices-precreate
+objects:
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: Role
+  metadata:
+    name: precreate-indices
+  rules:
+  - apiGroups:
+    - ""
+    resources:
+    - pods
+    verbs:
+    - list
+    - get
+  - apiGroups:
+    - ""
+    resources:
+    - pods/exec
+    verbs:
+    - create
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: RoleBinding
+  metadata:
+    name: precreate-indices
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: Role
+    name: precreate-indices
+  subjects:
+  - kind: ServiceAccount
+    name: ${CRON_SERVICE_ACCOUNT}
+    namespace: ${LOGGING_NAMESPACE}
+- apiVersion: batch/v1beta1
+  kind: CronJob
+  metadata:
+    name: precreate-indices
+    labels:
+       provider: openshift
+       logging-infra: indices-precreate
+  spec:
+    schedule: "${CRON_SCHEDULE}"
+    jobTemplate:
+      spec:
+        template:
+          metadata:
+            labels:
+              provider: openshift
+              logging-infra: indices-precreate
+          spec:
+            serviceAccount: ${CRON_SERVICE_ACCOUNT}
+            serviceAccountName: ${CRON_SERVICE_ACCOUNT}
+            containers:
+            - name: cli
+              image: ${CLI_IMAGE}
+              command: ["/bin/bash", "-c"]
+              args:
+              - echo "Starting pre-create of indices for Cluster Logging Elasticsearch...";
```

Comment:
Seems like moving this script to a separate file would be easier to maintain, no?

Reply:
It would, however, there is no easy way to provide that as a template which in this case is a 'one stop shop'. Ideally this is added into the product where it could then be moved into a script but its technically a feature enhancement so I'm certain it will live as a troubleshooting work around